### PR TITLE
raftstore: make threads of snap-generator pool configurable

### LINKED ISSUE #11247: Make threads of `snap-generator` pool be adjustable
## Background
In recover scenario,  because we want TiDB to make up the copy as soon as possible, we release all limit, including `store limit`, `scheduling limit`,  `snap-max-write-bytes-per-sec` and so on. but the IO only reach to 150MB/s for each TiKV instance. after we change the pool size of region worker from `2` to `32`, the recover speed has been effectively improved.
https://github.com/tikv/tikv/blob/14895dc46c48586946445ebff12e6380c7b561bf/components/raftstore/src/store/worker/region.rs#L631-L635

the IO speed reach to 650MB/s：
image

### PR DESCRIPTION
**Pull request by iosmanthus**
### What problem does this PR solve?

Issue Number: close #11247 

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- PR to update `pingcap/tidb-ansible`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

```release-note
make threads of snap-generator pool configurable
```

### CODE REVIEW

**Comment by iosmanthus** [comment]
@andylokandy PTAL

File: components/raftstore/src/store/config.rs [hunk 1]
```diff
@@ -143,7 +143,7 @@ pub struct Config {
 
     #[config(hidden)]
     pub use_delete_range: bool,
-
+    pub generate_pool_size: usize,
```
Comment (zhongzc) [inline components/raftstore/src/store/config.rs:146]: if it can not be online changed, please specify `#[config(skip)]`.

Comment (zhongzc) [reply components/raftstore/src/store/config.rs:146]: Doc is needed. Also for `config-template.toml`.

Comment (iosmanthus) [reply components/raftstore/src/store/config.rs:146]: DONE

Comment (nolouch) [reply components/raftstore/src/store/config.rs:148]: Do we have a pool that can support dynamically adjusting threads?

Comment (zhangjinpeng1987) [reply components/raftstore/src/store/config.rs:148]: snap_generating_pool_size

Comment (iosmanthus) [reply components/raftstore/src/store/config.rs:148]: @nolouch not yet. But the PR for raftstore's dynamic-sized thread pool is on the way.

**Review by nolouch — APPROVED** [review]

**Review by andylokandy — APPROVED** [review]

**Review by zz-jason — APPROVED** [review]
LGTM

**Comment by andylokandy** [comment]
@zhouqiang-cl PTAL

**Comment by iosmanthus** [comment]
@zhouqiang-cl PTAL

**Pull request closed without merging**